website: some stuff about workspace IDs

nfagerlund opened the pull request.

svanharmelen commented on website/docs/r/workspace.html.markdown:
```diff
@@ -53,10 +53,14 @@ The `vcs_repo` block supports:
+  and `tfe_variable` resources (and their related API calls).
+* `external_id` - The workspace's opaque external ID, which looks like
+  `ws-<RANDOM STRING>`. This is the ID format used by `tfe_sentinel_policy`
+  resources (and their related API calls).
```
Same about the listed resources.

nfagerlund commented on website/docs/d/workspace.html.markdown:
```diff
@@ -30,17 +30,19 @@ The following arguments are supported:
+  and `tfe_variable` resources (and their related API calls).
+* `external_id` - The workspace's opaque external ID, which looks like
+  `ws-<RANDOM STRING>`. This is the ID format used by `tfe_sentinel_policy`
+  resources (and their related API calls).
```
Both ID fields are non-configurable exported attributes... and this is the list specifically meant for non-configurable exported attributes. So I'm not sure what you mean?

svanharmelen replied:
Ah, sorry... This is the datasource, not the resource.

In that case I guess it makes less sense indeed. I do like to keep things ordered consistent and in the resource this is not a configurable field and so it is all the way at the bottom.

But indeed for the datasource that is different.

nfagerlund commented on website/docs/d/workspace_ids.html.markdown:
```diff
@@ -23,20 +28,20 @@ data "tfe_workspace" "test" {
-  Additionally you can also use a single entry with a wildcard (e.g. `"*"`) which
-  will match all names. Using a partial string together with a wildcard (e.g.
-  `"my-workspace-*"`) is **not** supported.
+    To select _all_ workspaces for an organization, provide a list with a single
```
Because `["*"]` isn't really a "special case." It's just that the `names` attribute accepts two different kinds of values: either a list of names, or a list of one asterisk. 

Whenever possible, the bullet point for an attribute should fully describe its allowed values, and putting the `~>` below the attribute list split that info into two different places.

svanharmelen replied:
Check... I took the approach from here: https://www.terraform.io/docs/configuration/resources.html#ignore_changes

But I guess this is fine as well. Yet if we just add this as additional text, should we then still include the blank line? Of just write it as one block?

nfagerlund replied:
Ah, gotcha! Well, for what it's worth, it looks like we [changed that for the 0.12 docs](https://github.com/hashicorp/terraform/blob/master/website/docs/configuration/resources.html.md#lifecycle-lifecycle-customizations). 

I tried a few ways to get it into a single paragraph, but they all made it harder to read and harder to skim. Compare:

> - `names` - (Required) Either a list of workspace names to search for, or a list with a single asterisk (`["*"]`) to request all workspaces. (No other use of wildcards is supported.) Names that don't match a real workspace will be omitted from the results, but are not an error.

svanharmelen replied:
Check 👍

nfagerlund commented:
I removed the resource lists, and explained why I moved the info about allowed values into the attribute's description — lmk if there's anything else needed!
